aws/auto-scaling-group/vpc.tf

```diff
@@ -19,21 +20,22 @@ resource "aws_internet_gateway" "igw" {
   })
 }
 
-resource "aws_subnet" "subnets" {
+# Private subnets in which the instances get private IP addresses.
+resource "aws_subnet" "instances_subnets" {
   count = length(var.availability_zones)
 
   vpc_id                  = aws_vpc.vpc.id
   cidr_block              = cidrsubnet(var.vpc_cidr_block, 4, count.index)
   availability_zone       = var.availability_zones[count.index]
   map_public_ip_on_launch = var.public_subnets # Default is false. 
-  # Subnets do not have public IPs per default.
 
   tags = {
     Name = "${var.name}-subnet-${var.availability_zones[count.index]}"
```

Comment:
QoL: I would prefer to not have the name of the resource inside each resource for two reasons.

1. We know that the resource names are a somewhat limited resource with a max of "only 24 chars allowed"
2. It feels duplicated. When I am listing the subnets, I don't think I need subnet inside the resource name for everything. Same for load balancer, security groups, etc.. It feels natural when developing this in terraform, but when we are inside the AWS console it just feels somewhat off. To illustrate this:
![grafik](https://github.com/THEY-Consulting/they-terraform/assets/44843935/260ed88f-5211-4bf7-8a73-40a1bf7d115e)

Could you adapt this for all resources? Perhaps if @thiemogries agrees? Should be a relatively quick "fix".

Reply:
I think this can be helpful in certain circumstances. I would keep it for resources, that are not limited to 24 chars

Reply:
I agree with @xddq, I removed the resource type from the name. Only in special cases where one would want to be able to differentiate between public and private resources I added `private` or `public` to the name, and the AZ where the resource is being deployed. Since I find it useful to know the AZ of a resource right away when I see all the resources in the AWS console.

I looked in the tf documentation of some resources, but I couldn't figure out in which ones there is this 24 chars limitation and in which there is no limitation 🙃